dali/ft/filecopy.cpp

```diff
@@ -3979,63 +4012,82 @@ bool FileSprayer::calcUsePull() const
         {
             StringBuffer s;
-            sources.item(idx2).filename.queryIP().getHostText(s);
-            LOG(MCdebugInfo, "Use pull operation because %s cannot push", s.str());
+            sources.item(0).filename.queryIP().getHostText(s);
+            LOG(MCdebugInfo, "Use pull operation because %s doesn't appear to have an ftslave", s.str());
             return true;
         }
-    }
-    if (!canLocateSlaveForNode(sources.item(0).filename.queryIP()))
-    {
-        StringBuffer s;
-        sources.item(0).filename.queryIP().getHostText(s);
-        LOG(MCdebugInfo, "Use pull operation because %s doesn't appear to have an ftslave", s.str());
-        return true;
-    }
 
-    ForEachItemIn(idx, targets)
-    {
-        if (!targets.item(idx).canPull())
+        ForEachItemIn(idx, targets)
+        {
+            if (!targets.item(idx).canPull())
+            {
+                StringBuffer s;
+                targets.item(idx).queryIP().getHostText(s);
+                LOG(MCdebugInfo, "Use push operation because %s cannot pull", s.str());
+                return false;
+            }
+        }
+
+        if (!canLocateSlaveForNode(targets.item(0).queryIP()))
         {
             StringBuffer s;
-            targets.item(idx).queryIP().getHostText(s);
-            LOG(MCdebugInfo, "Use push operation because %s cannot pull", s.str());
+            targets.item(0).queryIP().getHostText(s);
+            LOG(MCdebugInfo, "Use push operation because %s doesn't appear to have an ftslave", s.str());
             return false;
         }
     }
 
-    if (!canLocateSlaveForNode(targets.item(0).queryIP()))
+    bool pullRequested = options->hasProp(ANpull) ? options->getPropBool(ANpull) : false;
+    bool pushRequested = options->hasProp(ANpush) ? options->getPropBool(ANpush) : false;
+    if (!targetSupportsConcurrentWrite)
     {
-        StringBuffer s;
-        targets.item(0).queryIP().getHostText(s);
-        LOG(MCdebugInfo, "Use push operation because %s doesn't appear to have an ftslave", s.str());
-        return false;
+        // always pull
+        if (pushRequested)
+            IWARNLOG("Ignoring push option as targets < sources and target does not support concurrent write");
+        return true; // could be push if equal # of soruces and targets but no point
     }
-
-    //Use push if going to a single node.
-    if ((targets.ordinality() == 1) && (sources.ordinality() > 1))
+    else
     {
-        LOG(MCdebugInfo, "Use push operation because going to a single node from many");
-        return false;
-    }
+        bool wantPull = false;
+        bool wantPush = false;
+        if (targets.ordinality() < sources.ordinality()) // implying multiple writes to same target, force pull, and will common up on matching target filenames
+            wantPull = true;
+        else if (targets.ordinality() > sources.ordinality()) // targets > sources. i.e. multiple splits of source files for each target
```

Comment:
If doing an M to N copy (e.g. 20 to 30) you could still have multiple writes to the same target.  Probably an edge case we don't care about - since that is legal in this branch.

Reply:
this is when targetSupportsConcurrentWrite = true only (e.g. BM not cloud), so it's okay / desirable (or at least maintaining previous behaviour) to continue with push even if multiple sources being pushed to same target in this case?